### ISSUE TITLE
Unexpected keyword argument 'ggml_model'

### ISSUE THREAD

**Issue by trane293** [nuance1979/llama-server#9]
Following the instructions, when running:

```bash
llama-server --models-yml ./llama-30B-ggml/models.yml --model-id llama-30b 
```

I get the following error:

```bash
Exception ignored in: <function Model.__del__ at 0x102db7e50>
Traceback (most recent call last):
  File "/Users/asharma/.virtualenvs/llama-server-env/lib/python3.9/site-packages/pyllamacpp/model.py", line 336, in __del__
    if self._ctx:
AttributeError: 'Model' object has no attribute '_ctx'
Traceback (most recent call last):
  File "/Users/asharma/.virtualenvs/llama-server-env/bin/llama-server", line 8, in <module>
    sys.exit(main())
  File "/Users/asharma/.virtualenvs/llama-server-env/lib/python3.9/site-packages/click/core.py", line 1130, in __call__
    return self.main(*args, **kwargs)
  File "/Users/asharma/.virtualenvs/llama-server-env/lib/python3.9/site-packages/click/core.py", line 1055, in main
    rv = self.invoke(ctx)
  File "/Users/asharma/.virtualenvs/llama-server-env/lib/python3.9/site-packages/click/core.py", line 1404, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/Users/asharma/.virtualenvs/llama-server-env/lib/python3.9/site-packages/click/core.py", line 760, in invoke
    return __callback(*args, **kwargs)
  File "/Users/asharma/.virtualenvs/llama-server-env/lib/python3.9/site-packages/llama_server/server.py", line 169, in main
    globals()["model"] = Model(
TypeError: __init__() got an unexpected keyword argument 'ggml_model'
```

After looking at the https://abdeladim-s.github.io/pyllamacpp/ documentation, it turns out the argument name for `Model` function is indeed `model_path`, not `ggml_model`.

**Comment by trane293** [comment]
After fixing the source code to `model_path` instead of `ggml_model`, it worked, but the backend crashed with the following error:

```bash
INFO:     127.0.0.1:55226 - "GET /v1/models HTTP/1.1" 500 Internal Server Error
ERROR:    Exception in ASGI application
Traceback (most recent call last):
  File "/Users/asharma/.virtualenvs/llama-server-env/lib/python3.9/site-packages/uvicorn/protocols/http/httptools_impl.py", line 435, in run_asgi
    result = await app(  # type: ignore[func-returns-value]
  File "/Users/asharma/.virtualenvs/llama-server-env/lib/python3.9/site-packages/uvicorn/middleware/proxy_headers.py", line 78, in __call__
    return await self.app(scope, receive, send)
  File "/Users/asharma/.virtualenvs/llama-server-env/lib/python3.9/site-packages/fastapi/applications.py", line 276, in __call__
    await super().__call__(scope, receive, send)
  File "/Users/asharma/.virtualenvs/llama-server-env/lib/python3.9/site-packages/starlette/applications.py", line 122, in __call__
    await self.middleware_stack(scope, receive, send)
  File "/Users/asharma/.virtualenvs/llama-server-env/lib/python3.9/site-packages/starlette/middleware/errors.py", line 184, in __call__
    raise exc
  File "/Users/asharma/.virtualenvs/llama-server-env/lib/python3.9/site-packages/starlette/middleware/errors.py", line 162, in __call__
    await self.app(scope, receive, _send)
  File "/Users/asharma/.virtualenvs/llama-server-env/lib/python3.9/site-packages/starlette/middleware/exceptions.py", line 79, in __call__
    raise exc
  File "/Users/asharma/.virtualenvs/llama-server-env/lib/python3.9/site-packages/starlette/middleware/exceptions.py", line 68, in __call__
    await self.app(scope, receive, sender)
  File "/Users/asharma/.virtualenvs/llama-server-env/lib/python3.9/site-packages/fastapi/middleware/asyncexitstack.py", line 21, in __call__
    raise e
  File "/Users/asharma/.virtualenvs/llama-server-env/lib/python3.9/site-packages/fastapi/middleware/asyncexitstack.py", line 18, in __call__
    await self.app(scope, receive, send)
  File "/Users/asharma/.virtualenvs/llama-server-env/lib/python3.9/site-packages/starlette/routing.py", line 718, in __call__
    await route.handle(scope, receive, send)
  File "/Users/asharma/.virtualenvs/llama-server-env/lib/python3.9/site-packages/starlette/routing.py", line 276, in handle
    await self.app(scope, receive, send)
  File "/Users/asharma/.virtualenvs/llama-server-env/lib/python3.9/site-packages/starlette/routing.py", line 66, in app
    response = await func(request)
  File "/Users/asharma/.virtualenvs/llama-server-env/lib/python3.9/site-packages/fastapi/routing.py", line 237, in app
    raw_response = await run_endpoint_function(
  File "/Users/asharma/.virtualenvs/llama-server-env/lib/python3.9/site-packages/fastapi/routing.py", line 165, in run_endpoint_function
    return await run_in_threadpool(dependant.call, **values)
  File "/Users/asharma/.virtualenvs/llama-server-env/lib/python3.9/site-packages/starlette/concurrency.py", line 41, in run_in_threadpool
    return await anyio.to_thread.run_sync(func, *args)
  File "/Users/asharma/.virtualenvs/llama-server-env/lib/python3.9/site-packages/anyio/to_thread.py", line 31, in run_sync
    return await get_asynclib().run_sync_in_worker_thread(
  File "/Users/asharma/.virtualenvs/llama-server-env/lib/python3.9/site-packages/anyio/_backends/_asyncio.py", line 937, in run_sync_in_worker_thread
    return await future
  File "/Users/asharma/.virtualenvs/llama-server-env/lib/python3.9/site-packages/anyio/_backends/_asyncio.py", line 867, in run
    result = context.run(func, *args)
  File "/Users/asharma/.virtualenvs/llama-server-env/lib/python3.9/site-packages/llama_server/server.py", line 113, in models
    return ModelList(data=[ModelInfo(id=model_id)])
  File "pydantic/main.py", line 341, in pydantic.main.BaseModel.__init__
pydantic.error_wrappers.ValidationError: 1 validation error for ModelInfo
id
  none is not an allowed value (type=type_error.none.not_allowed)
```

**Comment by nuance1979** [comment]
This is because it can't find the model id `llama-30b` in your `models.yml` file. Try edit it to match your preferred name:

From:
```yaml
  llama-33b:
    name: LLAMA-33B
    path: 33B/ggml-model-q4_0.bin
```
To:
```yaml
 llama-30b:
    name: LLAMA-30B
    path: path/to/ggml-model-q4_0.bin
```

BTW, I don't know why people keep calling it `llama-30b` because if you read the original llama paper, it's "33B".

**Comment by nuance1979** [comment]
Another note: I tried to upgrade to `pyllamacpp==2.1.3` but the reply I got was completely garbage. I believe it's because `pyllamacpp` changed its behavior a lot recently. Try downgrade it by `pip install -U pyllamacpp==2.0.0`.

**Comment by nuance1979** [comment]
Fixed by #10
